src/code/cutscene_camera.c

```diff
@@ -353,28 +357,576 @@ void CutsceneCamera_Reset(void) {
+        camPos->y = (coeff[0] * pointCmd[sp44[0]].pos.y) + (coeff[1] * pointCmd[sp44[1]].pos.y) +
+                    (coeff[2] * pointCmd[sp44[2]].pos.y) + (coeff[3] * pointCmd[sp44[3]].pos.y);
+        camPos->z = (coeff[0] * pointCmd[sp44[0]].pos.z) + (coeff[1] * pointCmd[sp44[1]].pos.z) +
+                    (coeff[2] * pointCmd[sp44[2]].pos.z) + (coeff[3] * pointCmd[sp44[3]].pos.z);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/cutscene_camera/func_80162A50.s")
+    if (camFov != NULL) {
+        *camFov = (coeff[0] * miscCmd[sp44[0]].fov) + (coeff[1] * miscCmd[sp44[1]].fov) +
+                  (coeff[2] * miscCmd[sp44[2]].fov) + (coeff[3] * miscCmd[sp44[3]].fov);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/cutscene_camera/func_80162FF8.s")
+    if (camRoll != NULL) {
+        s16 sp3C[4];
+        s32 sp2C[3];
+        s32 temp;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/cutscene_camera/func_801631DC.s")
+        sp3C[0] = CAM_DEG_TO_BINANG(miscCmd[sp44[0]].roll);
+        sp3C[1] = CAM_DEG_TO_BINANG(miscCmd[sp44[1]].roll);
+        sp3C[2] = CAM_DEG_TO_BINANG(miscCmd[sp44[2]].roll);
+        sp3C[3] = CAM_DEG_TO_BINANG(miscCmd[sp44[3]].roll);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/cutscene_camera/func_80163334.s")
+        sp2C[0] = (s16)(sp3C[1] - sp3C[0]);
+        sp2C[1] = sp2C[0] + (s16)(sp3C[2] - sp3C[1]);
+        sp2C[2] = sp2C[1] + (s16)(sp3C[3] - sp3C[2]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/cutscene_camera/func_80163660.s")
+        temp = ((coeff[1] * sp2C[0]) + (coeff[2] * sp2C[1]) + (coeff[3] * sp2C[2]));
+        *camRoll = sp3C[0] + temp;
+    }
+
+    interpState->curFrame++;
+
+    if (interpState->curFrame == pointCmd[interpState->waypoint + 1].duration) {
+        interpState->curFrame = 0;
+        interpState->waypoint++;
+        if (interpState->waypoint >= (interpState->duration - 3)) {
+            interpState->type = CS_CAM_INTERP_OFF;
+            return interpState->duration;
+        }
+    }
+    return 0;
+}
+
+static f32 sKnots[38];
+
+// Only used by unused CutsceneCamera_Interp_Unused
+void func_80162FF8(s16 arg0) {
+    f32 var_fv0 = 0.0f;
+    s32 i;
+
+    sKnots[0] = 0.0f;
+    sKnots[1] = 0.0f;
+    sKnots[2] = 0.0f;
+
+    for (i = 3; i < arg0; i++) {
+        if (i == 3) {
+            var_fv0 += 0.9f;
+        } else if ((i == 4) || (i == (arg0 - 1))) {
+            var_fv0 += 0.6f;
+        } else {
+            var_fv0 += 0.3f;
+        }
+
+        sKnots[i] = var_fv0;
+    }
+
+    var_fv0 += 0.9f;
+    sKnots[i++] = var_fv0;
+    sKnots[i++] = var_fv0;
+    sKnots[i++] = var_fv0;
+}
+
+// Only used by unused CutsceneCamera_Interp_Unused
+void func_801631DC(f32 progress, s32 arg2, f32* coeff) {
+    f32 coeffTemp[4][4];
+    s32 i;
+    s32 j;
+    s32 k;
+
+    for (i = 0; i < 3 + 1; i++) {
+        for (j = 0; j < 3 + 1; j++) {
+            coeffTemp[i][j] = 0.0f;
+        }
+    }
```

Comment:
errrrr, why the 3+1?

Reply:
3 is the order of the interpolation (technically the i loop only needs to go to `< order` instead of `< order + 1`). (If you are curious the same pattern exists in `SubS_TimePathing_ComputeWeights`, but uses order as a variable so you can see it).

Reply:
I assume <=3 doesnt match?

Reply:
It does match actually. I might defer to @hensldm to see if that's better?

Reply:
Not sure I have a preference. Does it also work in SubS?

Reply:
Going from `(i < (numPoints - order + 1))` to `(i <= (numPoints - order))` does not match in SubS

Reply:
And the more direct example:
```c
for (i = 0; i < order; i++) {
    for (j = 0; j < order + 1; j++) {
        weightsTemp[i][j] = 0.0f;
    }
}
```
matches, but changing it to 
```c
for (i = 0; i < order; i++) {
    for (j = 0; j <= order; j++) {
        weightsTemp[i][j] = 0.0f;
    }
}
```
does not match

Reply:
Probably best to leave it as is than. Could possibly make a define for the order?

Reply:
Done, although could you double check it was used in the right places?

Reply:
Usages seem fine to me.